### PR TITLE
Broker client work - First PR

### PR DESCRIPTION
**Pull request by shoatman**
This has not been tested at all yet.  Getting the PR out such that the service side of the equation can potentially leverage what's been introduced for request, token and broker error responses.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerErrorResult.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.microsoft.identity.common.internal.broker;
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerErrorResult.java]: Minor: This class will need license info

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    public MicrosoftAuthServiceFuture connect() {
+
+        MicrosoftAuthServiceFuture future = new MicrosoftAuthServiceFuture();
+        mMicrosoftAuthServiceConnection = new MicrosoftAuthServiceConnection(future);
+
+        final boolean serviceBound = mContext.bindService(mMicrosoftAuthServiceIntent, mMicrosoftAuthServiceConnection, Context.BIND_AUTO_CREATE);
+        Logger.verbose(TAG + "connect", "The status for MicrosoftAuthService bindService call is: " + Boolean.valueOf(serviceBound));
+
+        if (!serviceBound) {
+            disconnect();
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: Judging from [the docs](https://developer.android.com/reference/android/content/Context.html#bindService(android.content.Intent,%20android.content.ServiceConnection,%20int)), we may not need to take this step if `false == serviceBound`

>`true` if the system is in the process of bringing up a service that your client has permission to bind to; `false` if the system couldn't find the service or if your client doesn't have permission to bind to it. If this value is `true`, you should later call unbindService(ServiceConnection) to release the connection.`

Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: Good catch.  I'll update this in a future PR.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            throw new RuntimeException("Unable to bind to service");
```
Comment (addev-ashish) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: based on our current struggles with boundService, crashing the app here might be problematic, we should look at throwing a client exception saying broker not reponding

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthServiceConnection.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        Logger.verbose(TAG, "MicrosoftAuthService is connected.");
+        mMicrosoftAuthService = IMicrosoftAuthService.Stub.asInterface(service);
+        mMicrosoftAuthServiceFuture.setMicrosoftAuthService(mMicrosoftAuthService);
+    }
+
+    /*
+
+    //Needed for API level 26 and above
+    @Override
+    public void onBindingDied(ComponentName name){
```
Comment (addev-ashish) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthServiceConnection.java]: this is interesting, can we add a log in these and run it with the Vivo device we have that has binding issues

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java (hunk 3 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: This is work in progress relative to exceptions.  The handler upstream can convert this to whatever i like.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthServiceConnection.java (hunk 4 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthServiceConnection.java]: I'll take a look.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java (hunk 2 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: Fixed it now.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerErrorResult.java (hunk 1 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerErrorResult.java]: Fixed it

File: common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java (hunk 3 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java]: I fixed this with clientException for now....

**Pull request merged**